src/main/java/pages/basepages/DetailsBase.java

```diff
@@ -19,11 +23,41 @@
     @CacheLookup
     protected WebElement editBtn;
 
-    public DetailsBase() {
+    @FindBy(css = "input[value='Delete']")
+    @CacheLookup
+    protected WebElement deleteBtn;
+
+    public DetailsBase(){
         this.driver = DriverManager.getInstance().getDriver();
         this.wait = DriverManager.getInstance().getWait();
         PageFactory.initElements(driver, this);
     }
 
     public abstract Object clickEditBtn();
+
+    public abstract Object clickDeleteBtn(boolean confirmDeletion);
+
+    public void clickDeleteButton(boolean confirmDeletion){
+        wait.until(ExpectedConditions.visibilityOf(deleteBtn));
+        deleteBtn.click();
+        Alert alert;
```

Comment:
Then we will need to move the implementation related to alert to webDriverTools class